music_this_week_app/backend/spotifyHandler.py

```diff
@@ -55,10 +60,12 @@ def filter_artist(self, artist_name):
             except spotipy.client.SpotifyException as error:
                 print("ERROR: Failed to search twice. Error below:")
                 print(error)
+                self.save_new_artist(artist_name, None)
```

Comment:
Do you really want these (this and the other one below) in the exceptions? I really should have commented this code better but both exceptions handle some sort of error on the Spotipy `search` call; the regular flow of "no matches found" routes through the `if num_matches == 0` if statement below.

Now that I look at it again, this method is pretty confusing (and now made more multi-purposed and less straightforward with the addition of this database functionality).

Could you please split it into several methods that do the same thing clearer?

```
filter_list_of_artists()
    filter_artists()
        lookup artist in database (add a comment to this block of code)
        search_spotify() # New method to actually send the search and handle exceptions
        filter_spotify_results() # New method to parse the Spotify results and come up with the correct URI
```